force-app/main/default/classes/ServiceParticipantRollupsService.cls

```diff
@@ -0,0 +1,74 @@
+/*
+ *
+ *  * Copyright (c) 2022, salesforce.com, inc.
+ *  * All rights reserved.
+ *  * SPDX-License-Identifier: BSD-3-Clause
+ *  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ *
+ */
+public with sharing class ServiceParticipantRollupsService {
+    private static String ENROLLED = 'Enrolled';
+    private static Set<String> FEATUREGATE_NAME = new Set<String>{
+        'ServiceParticipantsToServiceSchedule'
+    };
+
+    public void processParticipantRollups(List<ServiceParticipant__c> participants) {
+        Integer participantCount;
+        Set<Id> serviceScheduleIds = new Set<Id>();
+        List<ServiceSchedule__c> serviceSchedulesToUpdate = new List<ServiceSchedule__c>();
+        Map<Id, ServiceSchedule__c> schedules = new Map<Id, ServiceSchedule__c>();
+        Boolean isfeatureGateActive = getFeatureGateStatus();
+
+        if (isfeatureGateActive) {
+            for (ServiceParticipant__c participant : participants) {
+                serviceScheduleIds.add(participant.ServiceSchedule__c);
+            }
+
+            schedules = getServiceSchedules(serviceScheduleIds);
+
+            for (ServiceSchedule__c schedule : schedules.values()) {
+                participantCount = 0;
+                for (
+                    ServiceParticipant__c serviceParticipant : schedule.ServiceParticipants__r
+                ) {
+                    if (serviceParticipant.Status__c != ENROLLED) {
+                        continue;
+                    }
+                    participantCount++;
+                }
+                schedule.ParticipantsEnrolled__c = participantCount;
+                serviceSchedulesToUpdate.add(schedule);
```

Comment:
Maybe wrap this in an `if (schedule.ParticipantsEnrolled__c != participantCount) {}` to avoid updating any records where the field value isn't changing?